vercel/resource_project_rolling_release.go

```diff
@@ -0,0 +1,754 @@
+						Required:            true,
+					},
+					"advancement_type": schema.StringAttribute{
+						MarkdownDescription: "The type of advancement between stages. Must be either 'automatic' or 'manual-approval'. Required when enabled is true.",
+						Optional:            true,
+						Computed:            true,
+						Validators: []validator.String{
+							advancementTypeValidator{},
+						},
+					},
+					"stages": schema.ListNestedAttribute{
+						MarkdownDescription: "The stages of the rolling release. Required when enabled is true.",
+						Optional:            true,
+						Computed:            true,
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"target_percentage": schema.Int64Attribute{
+									MarkdownDescription: "The percentage of traffic to route to this stage.",
+									Required:            true,
+									Validators: []validator.Int64{
+										int64validator.Between(0, 100),
+									},
+								},
+								"duration": schema.Int64Attribute{
+									MarkdownDescription: "The duration in minutes to wait before advancing to the next stage. Required for all stages except the final stage when using automatic advancement.",
+									Optional:            true,
+								},
+								"require_approval": schema.BoolAttribute{
+									MarkdownDescription: "Whether approval is required before advancing to the next stage.",
+									Optional:            true,
+									Computed:            true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+type TFRollingReleaseStage struct {
+	TargetPercentage types.Int64 `tfsdk:"target_percentage"`
+	Duration         types.Int64 `tfsdk:"duration"`
+	RequireApproval  types.Bool  `tfsdk:"require_approval"`
+}
+
+// TFRollingRelease reflects the state terraform stores internally for a project rolling release.
+type TFRollingRelease struct {
+	Enabled         types.Bool   `tfsdk:"enabled"`
+	AdvancementType types.String `tfsdk:"advancement_type"`
+	Stages          types.List   `tfsdk:"stages"`
+}
+
+// ProjectRollingRelease reflects the state terraform stores internally for a project rolling release.
+type TFRollingReleaseInfo struct {
+	RollingRelease TFRollingRelease `tfsdk:"rolling_release"`
+	ProjectID      types.String     `tfsdk:"project_id"`
+	TeamID         types.String     `tfsdk:"team_id"`
+}
+
+func (e *TFRollingReleaseInfo) toUpdateRollingReleaseRequest() (client.UpdateRollingReleaseRequest, diag.Diagnostics) {
+	var stages []client.RollingReleaseStage
+	var advancementType string
+	var diags diag.Diagnostics
+
+	if e.RollingRelease.Enabled.ValueBool() {
+		// When enabled, both advancement_type and stages are required
+		if e.RollingRelease.AdvancementType.IsNull() || e.RollingRelease.AdvancementType.IsUnknown() {
+			diags.AddError(
+				"Error creating rolling release",
+				"advancement_type is required when enabled is true",
+			)
+			return client.UpdateRollingReleaseRequest{}, diags
+		}
```

Comment:
Rely on API validation here?

Reply:
removed in 92f0f0b662de6c343949bf394ec7be7859199f63